packages/jest-runner/src/index.ts

```diff
@@ -262,7 +244,13 @@ export default class TestRunner {
               ]);
             }
           })
-          .catch(error => onError(error, test)),
+          .catch(error => {
+            if (onFailure) {
+              return onFailure(test, error);
+            } else {
+              return this.eventEmitter.emit('test-file-failure', [test, error]);
+            }
```

Comment:
```suggestion

            return this.eventEmitter.emit('test-file-failure', [test, error]);
```

Reply:
Right. This pattern was repeating few times, so I refactored all of them.